#! /bin/sh /usr/share/dpatch/dpatch-run
Description: Check for and migrate old AqBanking configuration on first start
Author: Bill Nottingham <notting@redhat.com>
Origin: http://lists.gnucash.org/pipermail/gnucash-devel/2010-August/029265.html
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=593544
Applied-Upstream: http://lists.gnucash.org/pipermail/gnucash-changes/2010-August/008558.html

@DPATCH@

diff --git a/src/import-export/aqbanking/gnc-ab-utils.c b/src/import-export/aqbanking/gnc-ab-utils.c
index 0133c8a..1b43b01 100644
--- a/src/import-export/aqbanking/gnc-ab-utils.c
+++ b/src/import-export/aqbanking/gnc-ab-utils.c
@@ -120,6 +120,29 @@ gnc_AB_BANKING_new(void)
         api = AB_Banking_new("gnucash", NULL, 0);
         g_return_val_if_fail(api, NULL);
 
+#ifdef AQBANKING_VERSION_4_PLUS
+        /* Check for config migration */
+        if (AB_Banking_HasConf4(api, 0) != 0)
+        {
+            if (AB_Banking_HasConf3(api, 0) == 0)
+            {
+                g_message("gnc_AB_BANKING_new: importing aqbanking3 configuration\n");
+                if (AB_Banking_ImportConf3(api, 0) < 0)
+                {
+                    g_message("gnc_AB_BANKING_new: unable to import aqbanking3 configuration\n");
+                }
+            }
+            else if (AB_Banking_HasConf2(api, 0) == 0)
+            {
+                g_message("gnc_AB_BANKING_new: importing aqbanking2 configuration\n");
+                if (AB_Banking_ImportConf2(api, 0) < 0)
+                {
+                    g_message("gnc_AB_BANKING_new: unable to import aqbanking2 configuration\n");
+                }
+            }
+        } 
+#endif /* AQBANKING_VERSION_4_PLUS */        
+        
         /* Init the API */
         g_return_val_if_fail(AB_Banking_Init(api) == 0, NULL);
 
